src/android/interoperability/cpp/android-build-rust.md

```diff
@@ -0,0 +1,12 @@
+# Building in Android
+
+Create a `rust_binary` that depends on `libcxx` and your `cc_library_static`.
+
+```json
```

Comment:
I used `javascript` as the language in the Android part (should be mostly the same, so I'm just mentioning it for consistency).

I also used a particular format for referring to external files, see [Rust Binaries](https://google.github.io/comprehensive-rust/android/build-rules/binary.html). 
I did that consistently in those slides: this was important since I hope people can copy-paste the code into their own slides and thus follow-along at home.  To make my life easy, all the files are then actual live files so that I can navigate to the right directory and do `m hello_rust_with_dep`.

See [build_all.sh](https://github.com/google/comprehensive-rust/blob/main/src/android/build_all.sh) for the script that I use to build everything once in a while. You should try to hook up your `Android.bp` files to the same script so that we can prevent bit-rot.

Reply:
I just created #1434 to explain this process a bit — it's a little hacky, but I think it's better than nothing :grimacing:

Reply:
> I also used a particular format for referring to external files, see [Rust Binaries](https://google.github.io/comprehensive-rust/android/build-rules/binary.html).

I'm not sure what you're referring to here? What "format" are you referring to? Do you mean the way that you're `#include`ing the snippets in from a larger example, whereas I'm writing most of them inline?

Reply:
> I'm not sure what you're referring to here? What "format" are you referring to?

Sorry, I was thinking of the 

````markdown

_filename.rs_:

```javascript

```
````

convention with the italic filename. If the precise name doesn't matter, then don't worry about following it (I found that the names matter a lot for the care of the AIDL code).